packages/web-components/src/components/ic-dialog/ic-dialog.tsx

```diff
@@ -0,0 +1,385 @@
+    this.dialogRendered = true;
+    setTimeout(() => {
+      this.fadeIn = true;
+    }, 10);
+    setTimeout(() => {
+      this.setInitialFocus();
+    }, 75);
+    setTimeout(() => {
+      this.getFocusedElementIndex();
+    }, 80);
+    document.body.style.height = "100%";
+    document.body.style.overflow = "hidden";
+  }
+  /**
+   * Used to dismiss the dialog.
+   */
+  @Method()
+  async hideDialog() {
+    this.fadeIn = false;
+    setTimeout(() => {
+      this.dialogRendered = false;
+      this.sourceElement.focus();
+      document.body.style.height = "auto";
+      document.body.style.overflow = "auto";
+    }, 80);
+  }
+
+  @Listen("keydown", { target: "document" })
+  handleKeyboard(ev: KeyboardEvent): void {
+    if (this.dialogRendered) {
+      switch (ev.key) {
+        case "Tab":
+          ev.preventDefault();
+          this.focusNextInteractiveElement(ev.shiftKey);
+          break;
+        case "Escape":
+          !ev.repeat && this.hideDialog();
+          ev.stopImmediatePropagation();
+          break;
+      }
+    }
+  }
+
+  @Listen("click")
+  handleClick(ev: MouseEvent): void {
+    const dialogElement = this.el.shadowRoot.querySelector("dialog");
+    if (this.closeOnClick) {
+      if (ev.composedPath().indexOf(dialogElement) < 0) {
+        this.hideDialog();
+      }
+    }
+  }
+
+  private setInitialFocus = () => {
+    this.sourceElement = document.activeElement as HTMLElement;
+
+    let focusedElement;
+
+    if (this.el.querySelector("#focus") !== null) {
+      focusedElement = this.el.querySelector("#focus") as HTMLElement;
+    } else {
+      focusedElement = this.el.shadowRoot.querySelector(
+        "#focus"
+      ) as HTMLElement;
+    }
+
+    if (process.env.JEST_WORKER_ID === undefined) {
+      focusedElement.focus();
+    }
+  };
+
+  private getFocusedElementIndex = () => {
+    for (let i = 0; i < this.interactiveElementList.length; i++) {
+      if (
+        (this.interactiveElementList[i] as HTMLElement) ===
+        (this.el.shadowRoot.activeElement || document.activeElement)
+      ) {
+        this.focusedElementIndex = i;
+      }
+    }
+  };
+
+  private setAlertVariant = () => {
+    if (this.status !== undefined) {
+      const alert = this.el.shadowRoot.querySelector("ic-alert");
+      alert.setAttribute("variant", this.status);
+    }
+  };
+
+  private getInteractiveElements = () => {
+    this.interactiveElementList = Array.prototype.slice.call(
+      this.el.shadowRoot.querySelectorAll("ic-button")
+    );
+    const slottedInteractiveElements = Array.prototype.slice.call(
+      this.el.querySelectorAll(
+        `a[href], button, input, textarea, select, details, [tabindex]:not([tabindex="-1"]), 
+        ic-button, ic-checkbox, ic-select, ic-search-bar, ic-tab-group, ic-radio-group, 
+        ic-back-to-top, ic-breadcrumb, ic-chip[dismissible="true"], ic-footer-links, ic-link, ic-navigation-button, 
+        ic-navigation-item, ic-switch`
+      )
+    );
```

Comment:
Is the `slice` array method required here? It looks like we just want to put a list of items into an array?